### ISSUE TITLE
Setting VF_SHARE_LAYERS to False and NO_FINAL_LINEAR to true leads to a bug

### ISSUE THREAD

**Issue by pouyahmdn** [ray-project/ray#21756]
### Search before asking

- [X] I searched the [issues](https://github.com/ray-project/ray/issues) and found no similar issues.

### Ray Component

RLlib

### What happened + What you expected to happen

If you set 'vf_share_layers' to False and 'no_final_linear' to True, the value function in [fcnet.py](https://github.com/ray-project/ray/blob/b6d3e01e0b5928fa3cfb747c0b7c2ad65f0e3f60/rllib/models/torch/fcnet.py#L135) leads to an error. The error is because the [final layer](https://github.com/ray-project/ray/blob/b6d3e01e0b5928fa3cfb747c0b7c2ad65f0e3f60/rllib/models/torch/fcnet.py#L105) of `self._value_branch_separate` is not equal in size to the [first layer](https://github.com/ray-project/ray/blob/b6d3e01e0b5928fa3cfb747c0b7c2ad65f0e3f60/rllib/models/torch/fcnet.py#L109) of 'self._value_branch'. The variables `prev_vf_layer_size` and `prev_layer_size` are not equal. `prev_vf_layer_size` is equal to `hiddens[-1]` and `prev_layer_size` is equal to `num_outputs`.

### Versions / Dependencies

The bug should not depend on anything.

### Reproduction script

```
import ray
from ray.rllib.agents import ppo
import some_env


def main():
    ray.init()
    config_ray = ppo.DEFAULT_CONFIG.copy()
    config_ray["framework"] = "torch"

    config_ray["model"]["vf_share_layers"] = False
    config_ray["model"]["no_final_linear"] = True

    trainer = ppo.PPOTrainer(env= some_env)

    result = trainer.train()
    trainer.save()
    ray.shutdown()


if __name__ == '__main__':
    main()

```

### Anything else

Always.

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

**Comment by gresavage** [comment]
I'm running into this issue as well.

**Comment by cszhu** [comment]
Hello! This P2 issue has seen no activity in at least 2 years. It will be closed in 2 weeks as part of ongoing cleanup efforts.

Please remove the `pending-cleanup` label if you believe this issue should remain open.

Thanks for contributing to Ray!
